formal_spec/README.md

```diff
@@ -0,0 +1,95 @@
+The folder `formal_spec` contains a formal specification of the NMT proof/verification logic in [Quint](https://github.com/informalsystems/quint).
```

Comment:
[CIs failing] Could you please move this line to another location in the document? Our CIs require that a markdown file starts with a top-level heading, and this line is currently causing the CIs to fail. Thank you!